### PR TITLE
feat: Distributed Authority Sample Avatar interactions (pick up & throw) [MTT-8905] [MTT-8884]

### PR DESCRIPTION
**Pull request by fernando-cortez**
### Description
This PR adds the pickup, drop & toss functionality for an avatar interacting with a `Pickupable` object.

Inside HubScene:
* spawners have been added to instantiate (as session owner) a Distributable & Transferable NetworkObject (a simple cube for now), 6 in total

To validate:
* connect to service, approach a cube, and press interact button (E)
* if button was tapped, the avatar will change ownership of interacted cube, and through FixedJoints fix the cube to the avatar
* once held, one can: 
  * tap interact button again to simply drop cube
  * if interact is held (> 0.5s) the cube will be thrown

Caveat: Once picked up, cube NetworkObjects are marked as RequestRequired. This means that one client can approach another client's held cube, request ownership transfer, and become the new holding owner of said cube. 

### Issue Number(s)
[MTT-8905](https://jira.unity3d.com/browse/MTT-8905) & [MTT-8884](https://jira.unity3d.com/browse/MTT-8884)

### Contribution checklist
 - [ N/A ] Tests have been added for the project and/or any internal package
 - [ N/A ] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] JIRA ticket ID is in the PR title or at least one commit message
 - [x] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

**Comment by NoelStephensUnity** [comment]
I have this opened and in my to review queue/window.

File: Experimental/DistributedAuthoritySample/Assets/Scripts/Gameplay/EnvironmentTransform.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        protected override void OnOwnershipChanged(ulong previous, ulong current)
+        {
+            base.OnOwnershipChanged(previous, current);
+
+            OnGameplayEvent?.Invoke(NetworkObject, GameplayEvent.OwnershipChange);
+        }
+
+        // note: invoked on owning client
+        bool OnOwnershipRequested(ulong clientRequesting)
```
Comment (NoelStephensUnity) [inline Experimental/DistributedAuthoritySample/Assets/Scripts/Gameplay/EnvironmentTransform.cs:41]: The only very minor nit-pick type of thing would be adding `private` in front of some of the methods and properties throughout the various classes.

**Review by NoelStephensUnity — APPROVED** [review]
It all looks good. 
Minor comment about sprinkling `private` in front of some of the methods and properties (even though it is implied).

**Comment by fernando-cortez** [comment]
Merging this in, but [this task](https://jira.unity3d.com/browse/MTT-9016) tracks the v2.0 version of this pickup task, that implements the FixedJoint approach in an inverse way.

**Pull request merged**